src/parser/on_demand/parse_to_structure.ts

```diff
@@ -0,0 +1,138 @@
+import { type Code } from '../../code';
+import { type BSONElement, getSize, parseToElements as p } from './parse_to_elements';
+
+/** @internal TODO */
+const DEFAULT_REVIVER = () => null;
+
+/** @internal */
+function parseToElements(...args: Parameters<typeof p>): BSONElement[] {
+  const res = p(...args);
+  return Array.isArray(res) ? res : [...res];
+}
+
+/**
+ * @internal
+ * BSONElement offsets
+ */
+const enum e {
+  type = 0,
+  nameOffset = 1,
+  nameLength = 2,
+  offset = 3,
+  length = 4
+}
+
+/**
+ * @internal
+ * Embedded bson types
+ */
+const enum t {
```

Comment:
Even thought there are API docs, can we have more descriptive names for `e` and `t`, something like `elementOffsets` and `embeddedTypes`, respectively. Especially since the embedded bson types const could easily be confused for `e.`

Reply:
I would like to keep these as single characters as we did in parseToElements. I think the context matters, `it[t.object]` does not make much sense for what `it` is in the context of the function below. If this was exported for use elsewhere in the library then I think you would want the variables to carry more context in their names. I moved them closer to the point in which they are used to get better association with the code that uses them. Is that a good enough way to improve the meaning?

Reply:
Ah I see, sure, in the case that we ever do export them, we can change their names then.

Reply:
@baileympearson It seems like you have a similar suggestion as I did, here's context for the discussion we had before.

Reply:
I don't see an occurrence of `it[t.object]` in the code.  I do see the following:

```typescript
      const isEmbeddedType =
        type === t.object || type === t.array || type === t.javascriptWithScope;
```

Which makes sense with `EmbeddedTypes` instead of `t`.

I see something similar for `e`:
```typescript
      const type = it[e.type];
      const offset = it[e.offset];
```
but isn't that more clear if we just give it a good name?
```typescript
      const type = it[RawBSONElementIndexes.type];
      const offset = it[RawBSONElementIndexes.offset];
```

---

But more generally - I don't understand the arguments against descriptive variable names.  If we merged `parseToElements` that way, I suggest we fix it instead of forging ahead with unhelpful variable names.  

What downside is there to using good variable names?  Generally the only argument I hear against them is that it increases code size, but anyone in the JS community who cares about this bundles their code and has symbol minification anyways.

Reply:
Changed to desciptive names, thanks for reopening the thread Aditi so we could get our thoughts in one place. I was just going with my stylistic choice, but no need to stick to it. There is no increase to code size.

Reply:
LGTM, I'll wait for Bailey's response to close this thread.

Reply:
LGTM